app/src/main/java/org/schabi/newpipe/ReCaptchaActivity.java

```diff
@@ -137,23 +178,54 @@ private void saveCookiesAndFinish() {
     }
 
 
-    private void handleCookies(final String url) {
+    private void handleCookiesFromUrl(final @Nullable String url) {
+        if (MainActivity.DEBUG) {
+            Log.d(TAG, "handleCookiesFromUrl: url=" + (url == null ? "null" : url));
+        }
+
+        if (url == null) {
+            return;
+        }
+
         String cookies = CookieManager.getInstance().getCookie(url);
+        handleCookies(cookies);
+
+        // sometimes cookies are inside the url
+        int abuseStart = url.indexOf("google_abuse=");
+        if (abuseStart != -1) {
+            int abuseEnd = url.indexOf("+path");
+
+            try {
+                String abuseCookie = url.substring(abuseStart + 13, abuseEnd);
+                abuseCookie = URLDecoder.decode(abuseCookie, "UTF-8");
+                handleCookies(abuseCookie);
+            } catch (UnsupportedEncodingException | StringIndexOutOfBoundsException e) {
+                if (MainActivity.DEBUG) {
+                    e.printStackTrace();
+                    Log.d(TAG, "handleCookiesFromUrl: invalid google abuse starting at "
+                            + abuseStart + " and ending at " + abuseEnd + " for url " + url);
+                }
+            }
+        }
+    }
+
+    private void handleCookies(final @Nullable String cookies) {
         if (MainActivity.DEBUG) {
-            Log.d(TAG, "handleCookies: "
-                    + "url=" + url + "; cookies=" + (cookies == null ? "null" : cookies));
+            Log.d(TAG, "handleCookies: cookies=" + (cookies == null ? "null" : cookies));
         }
+
         if (cookies == null) {
             return;
         }
 
         addYoutubeCookies(cookies);
-        // add other methods to extract cookies here
+        // add here methods to extract cookies for other services
     }
 
-    private void addYoutubeCookies(@NonNull final String cookies) {
+    private void addYoutubeCookies(final @NonNull String cookies) {
```

Comment:
Why this change? We have `final` after the `@`s everywhere…

Reply:
I don't know ;-)
Fixed